### ISSUE TITLE
Image sizes should only be fixed on one dimension

### ISSUE THREAD

**Issue by kentcb** [dnfield/flutter_svg#227]
Consider the following repro, which I have attached in full as a zip at the end of this issue:

```dart
import 'package:flutter/material.dart';
import 'package:flutter_svg/flutter_svg.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    final column = Column(
      children: <Widget>[
        Row(children: [
          ImageWithText(),
        ]),
        Row(
          children: <Widget>[
            ImageWithText(),
            ImageWithText(),
          ],
        ),
        Row(
          children: <Widget>[
            ImageWithText(),
            ImageWithText(),
            ImageWithText(),
            ImageWithText(),
          ],
        ),
        Row(
          children: <Widget>[
            ImageWithText(),
            ImageWithText(),
            ImageWithText(),
            ImageWithText(),
            ImageWithText(),
            ImageWithText(),
          ],
        ),
        Row(
          children: <Widget>[
            ImageWithText(),
            ImageWithText(),
            ImageWithText(),
            ImageWithText(),
            ImageWithText(),
            ImageWithText(),
            ImageWithText(),
            ImageWithText(),
          ],
        ),
      ],
    );
    final scaffold = Scaffold(
      appBar: AppBar(),
      body: column,
    );
    final materialApp = MaterialApp(
      home: scaffold,
    );
    return materialApp;
  }
}

class ImageWithText extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    final image = SvgPicture.asset('assets/sample.svg');
    //final image = Image.asset('assets/bitmap.png');
    final imageContainer = Container(
      child: image,
      color: Colors.amber,
    );
    final text = Text('Hello');
    final column = Column(
      children: <Widget>[
        imageContainer,
        text,
      ],
    );
    final expanded = Expanded(
      child: column,
    );
    return expanded;
  }
}
```

When displaying a bitmap (via `Image.asset`) I see:

![Screenshot_20190923-165832](https://user-images.githubusercontent.com/1901832/65406656-8b12d680-de23-11e9-9f4f-b5bce70ffc28.png)

But when displaying an SVG (via `SvgPicture.asset`) I see:

![Screenshot_20190923-165851](https://user-images.githubusercontent.com/1901832/65406658-8e0dc700-de23-11e9-8b75-c1fc0254b0ca.png)

Notice how as the horizontal space gets tighter, the images correctly get narrower, but the SVGs incorrectly retain their height as they get narrower.

Debugging through the code, I believe the problem is [here](https://github.com/dnfield/flutter_svg/blob/master/lib/svg.dart#L712-L713). Setting both the width and height means that the `SizedBox` only shrinks in one direction. I think it should be setting whichever dimension is largest. I tried commenting out the assignment to `height` (which works for this case) and it worked as expected. I think the code should be something like:

```dart
if (width == null && height == null) {
    width = viewport.width > viewport.height ? viewport.width : null;
    height = viewport.height > viewport.width ? viewport.height : null;
} 
```

[repro_svg_sizing_issue.zip](https://github.com/dnfield/flutter_svg/files/3640939/repro_svg_sizing_issue.zip)

**Comment by fl0cke** [comment]
I agree that box sizing is broken at the moment, but removing the SizedBox as in #228 is not a valid fix as it breaks many other use cases, especially the automatic scaling and fitting of the image.

**Comment by ramsestom** [comment]
any update on this issue? kind of an annoying bug

**Comment by dnfield** [comment]
Argh. This ended up being much more breaking than anticipated. I'm probably going to have to at least temporarily revert it. Looking into that more later though.

**Comment by dnfield** [comment]
Remaining issue fixed by #529
